govtool/frontend/src/pages/PublicRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import { FC, ReactNode } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+
+import { checkIsWalletConnected } from "../utils";
+
+export const PublicRoute: FC<{ children: ReactNode }> = ({ children }) => {
+  // checkIsWalletConnected returns true if wallet is NOT connected
```

Comment:
Good catch - that is indeed counter intuitive. I would rather replace the condition in this helper (or at least name of that util) instead of negating with comment.

Could you please update that and provide change in CHANGELOG.md?